code/modules/admin/verbs/mapping.dm

```diff
@@ -517,3 +518,65 @@ GLOBAL_VAR_INIT(say_disabled, FALSE)
 		to_chat(usr, "Atmos control frequency check passed without encountering problems.", confidential=TRUE)
 	else
 		to_chat(usr, "Total errors: [invalid_machine + invalid_tag + tagless + duplicate_tag + not_heard + not_told]", confidential=TRUE)
+
+/// Check all tiles with a vent or scrubber on it and ensure that nothing is covering it up.
+/client/proc/check_for_obstructed_atmospherics()
+	set name = "Check For Obstructed Atmospherics"
+	set category = "Mapping"
+	if(!src.holder)
+		to_chat(src, "Only administrators may use this command.", confidential = TRUE)
+		return
+	SSblackbox.record_feedback("tally", "admin_verb", 1, "Check For Obstructed Atmospherics") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+	var/list/results = list()
+
+	results += "<h2><b>Anything that is considered to aesthetically obstruct an atmospherics machine (vent, scrubber, port) is listed below.</b> Please re-arrange to accomodate for this.</h2><br>"
+
+	// Ignore out stuff we see in normal and standard mapping that we don't care about (false alarms). Typically stuff that goes directionally off turfs or are undertiles that shouldn't matter.
+	var/list/ignore_list = list(
```

Comment:
cant you replace like half of those with checking for density

Reply:
It'll miss out on plants and the like, which is something that I do want to get rid of.

Reply:
I've been grappling with this for the last hour and I can't figure out how to pass a density check though without breaking the later logic (since obstructions are stored in a list), and every time I try to add a "for" check, it just owns itself on compile. I kept trying to do something like:

```dm
	for(var/turf/T in world)
		var/obj/machinery/atmospherics/components/unary/device = locate() in T.contents
		if(device)
			var/list/obj/obstruction_list = locate(/obj) in T.contents

			for(var/obj/obstruction in obstruction_list)
				if(!obstruction.density == !istype(obstruction, /obj/item))
					return

				if(!is_type_in_list(obstruction, ignore_list))
					results += "There is an obstruction on top of an atmospherics machine at: [ADMIN_VERBOSEJMP(T)].<br>"
					continue
```

but the verb just completely breaks. How do I check density on the `/obj`'s inside `obstruction_list`?